JSIL/ILBlockTranslator.cs

```diff
@@ -1130,11 +1138,28 @@ bool forDynamic
             var translated = TranslateNode(node as dynamic);
 
             var statement = translated as JSStatement;
-            if (statement == null) {
-                var expression = (JSExpression)translated;
+            if (statement == null)
+            {
+                var expression = (JSExpression) translated;
 
                 if (expression != null)
+                {
+                    var ranges =
```

Comment:
Probably it would be better always preserve source ILRange in  JSNode on this stage, and introduce separate pass over converted AST representation to build SourceMap if needed, as this processing could be really more complicated.
Right now JSIL may reuse JSNode (as variables) - and it may be a problem for this task. Will investigate a little bit more proper point of ILTranslation extension.
Will be good if right solution would not require us rewrite each JSNode creation.